internal/output/html.go

```diff
@@ -0,0 +1,486 @@
+package output
+
+import (
+	"embed"
+	"fmt"
+	"html/template"
+	"io"
+	"slices"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/google/osv-scanner/internal/identifiers"
+	"github.com/google/osv-scanner/internal/semantic"
+	"github.com/google/osv-scanner/internal/utility/severity"
+	"github.com/google/osv-scanner/pkg/models"
+)
+
+type HTMLResult struct {
+	HTMLVulnCount    HTMLVulnCount
+	EcosystemResults []EcosystemResult
+}
+
+type EcosystemResult struct {
+	Ecosystem string
+	Sources   []SourceResult
+}
+
+type SourceResult struct {
+	Source         string
+	Ecosystem      string
+	PackageResults []PackageResult
+	PackageCount   [2]int // called and uncalled package count
+	HTMLVulnCount  HTMLVulnCount
+}
+
+type PackageResult struct {
+	Name             string
+	Ecosystem        string
+	Source           string
+	CalledVulns      []HTMLVulnResult
+	UncalledVulns    []HTMLVulnResult
+	InstalledVersion string
+	FixedVersion     string
+	HTMLVulnCount    HTMLVulnCount
+}
+
+type HTMLVulnResult struct {
+	Summary HTMLVulnResultSummary
+	Detail  map[string]string
```

Comment:
As discussed offline, rather than looping through the fields, just list them out explicitly here.

Reply:
Updaetd with a `HTMLVulnResultDetail` struct